src/onepay-direct-qr.js

```diff
@@ -16,23 +23,17 @@ class Onepay {
       console.log('ott does not exist in object param');
       return;
     }
-    this.qrCodeAsBase64 = transaction.qrCodeAsBase64;
-  }
 
-  drawQrImage(htmlTagId) {
-    const instance = this;
-    let socketSubscribePromise = new Promise((resolve) => {
-      let socket = new OnepayWebSocket(instance.transaction);
+    this.qrCodeAsBase64 = this.transaction.qrCodeAsBase64;
 
-      socket.connect(() => resolve());
-    });
+    let socket = new OnepayWebSocket(this.transaction);
 
-    socketSubscribePromise.then(function () {
+    const instance = this;
+    socket.connect(function () {
```

Comment:
I think you can remove the ```const instance = this; ``` line and use ```this.qrCodeAsBase64``` below if you change the function() {} callback into an () => {} callback

Reply:
I will make a test and if it works i will change it.

Reply:
It works :)

Reply:
Great!